contents/docs/user-guides/sso.md

```diff
@@ -95,3 +105,129 @@ To set up Google SSO please follow these instructions:
+        <td>Environment variable</td>
+        <td>Optional?</td>
+    </tr>
+    <tr>
+        <td>Permanent ID</td>
+        <td><code>name_id</code></td>
+        <td><code>SAML_ATTR_PERMANENT_ID</code></td>
+        <td>❌ No</td>
+    </tr>
+    <tr>
+        <td>Email</td>
+        <td><code>email</code></td>
+        <td><code>SAML_ATTR_EMAIL</code></td>
+        <td>❌ No</td>
+    </tr>
+    <tr>
+        <td>First (Given) Name</td>
+        <td><code>first_name</code></td>
+        <td><code>SAML_ATTR_FIRST_NAME</code></td>
+        <td>❌ No</td>
+    </tr>
+    <tr>
+        <td>Last Name (Surname)</td>
+        <td><code>last_name</code></td>
+        <td><code>SAML_ATTR_LAST_NAME</code></td>
+        <td>✅ Yes</td>
+    </tr>
+</table>
+
+If your IdP does not send these attributes based on the default name on PostHog you have two options. Either adjust your IdP configuration to send the attributes with PostHog's default name, or set the respective environment variable (as noted above) to the attribute name your provider uses.
+
+3. You will now need to obtain some parameters from your IdP and configure them in PostHog as [environment variables][env-vars]. Depending on your provider, they may only provide this information as an XML metadata file. If this is the case, you can open the file in a text editor and obtain the required values from there.
+    - `SAML_ENTITY_ID`: Will be identified as EntityID or IdP issuer. This can vary greatly between providers, but it usually looks like a URL.
+    - `SAML_ACS_URL`: The endpoint to which the SAML requests are posted. It's usually called SAML endpoint  or IdP sign-on URL.
+    - `SAML_X509_CERT`: The public certificate used to validate SAML assertions from your IdP. It must be in X509 (almost all providers will provide it in this format). If your provider gives you the certificate as a file (usually `.pem`), just open it with a text editor to get the contents. When setting this certificate be sure to keep any spaces and new lines (don't format it). The first and last line of the certificate (`-----BEGIN CERTIFICATE-----`) are optional.
+
+4. Once you've configured all the settings above, you can now visit `<yourdomain>/login/saml/?idp=posthog_custom` (note the trailing slash and the query string) and you should be redirected to your IdP's login page. Upon completion, you'll find yourself in PostHog, fully authenticated.
+
+5. For security reasons, we don't output errors directly in your browser when something goes wrong. If you get an error in the process and you need to debug on PostHog's side you have two options:
+    - **Recommended**. Check your app logs (this varies depending on your deployment). Any errors will be logged there.
+    - If everything else fails, **temporarily** set environment variable `DEBUG = 1`, errors will be fully displayed now in the browser. **Please be sure to remove this once you're done, ugly things can happen if you don't.**
```

Comment:
How ugly are the things? 🤔

Reply:
I'd use `DEBUG=1` so I can copy/paste this to any tool where whitespace might be an issue